### DISCUSSION TITLE
how to get the length of spline?

### DISCUSSION THREAD

**Discussion by XiangjiBU** [mozman/ezdxf#650] [Q&A]
how to get the length of spline generated by add_spline(). for example:

```
fit_points=[(0,0,0),(750,500,0),(1750,500,0),(2250,1250,0)]
msp = doc.modelspace()
spline=msp.add_spline(fit_points)

```

**Comment by mozman** [comment]
There exist no analytical function to calculate the length of a B-spline, you have to approximate the curve and calculate the length  of the polyline.

**Comment by XiangjiBU** [comment]
Could you give me some guidance to get this? 
do you mean that I need to get all the knots, then approximate the length?
how can I get all the knots?  Is there any library can do this approximate?


On 3/11/2022 00:45，Manfred ***@***.***> wrote：

There exist no analytical function to calculate the length of a B-spline, you have to approximate the curve and calculate the length of the polyline.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Reply by mozman** [reply]
```python
import ezdxf
from ezdxf.math import ConstructionPolyline

doc = ezdxf.new()
msp = doc.modelspace()
fit_points = [(0, 0, 0), (750, 500, 0), (1750, 500, 0), (2250, 1250, 0)]

spline = msp.add_spline(fit_points)
# Adjust the max. sagitta distance to your needs or run the calculation in a loop 
# reducing the distance until the difference to the previous run is smaller 
# than your expected precision:
polyline = ConstructionPolyline(spline.flattening(distance=0.1))
print(f"approximated length = {polyline.length:.2f}")

```
